ui/src/router/components/form/components/router_config/GeneralSettingsPanel.js

```diff
@@ -60,6 +70,59 @@ export const GeneralSettingsPanel = ({
 
         <EuiSpacer size="m" />
 
+        <EuiFormRow
+          fullWidth
+          label={
+            <FormLabelWithToolTip
+              label="Protocol *"
+              content="Specify the type of Turing router Protocol"
+            />
+          }
+          display="row"
+        >
+          <EuiSuperSelect
+            fullWidth
+            options={protocolTypeOptions}
+            valueOfSelected={protocol}
+            onChange={(e) => {
+              onChange("config.protocol")(e);
```

Comment:
Nit: Can we define this handler separately so it can just be used here like `onChange={onChangeProtocol}` ? It more readable that way.